configuration/rules-blockly-timers-and-delays.md

```diff
@@ -0,0 +1,255 @@
+```javascript
+if (typeof this.timers['nothingTimerBlock'] === 'undefined' || this.timers['nothingTimerBlock'].hasTerminated()) {
+  this.timers['nothingTimerBlock'] = scriptExecution.createTimer(zdt.now().plusSeconds(10), function () {
+    })
+} else {
+  // do nothing
+}
+```
+In the case of *cancel* the `else{}` branch contains code to cancel the timer.
+
+```javascript
+if (typeof this.timers['cancelTimerBlock'] === 'undefined' || this.timers['cancelTimerBlock'].hasTerminated()) {
+  this.timers['cancelTimerBlock'] = scriptExecution.createTimer(zdt.now().plusSeconds(10), function () {
+    })
+} else {
+  this.timers['cancelTimerBlock'].cancel();
+  this.timers['cancelTimerBlock'] = undefined;
+}
+```
+In the case of *reschedule* the `else{}` statement contains code to reschedule the timer - restart the countdown. In the example generated code below:
+
+* Imagine the rule is triggered at 0 elapsed seconds.
+* The timer is started with a 10 second countdown.
+* Lets say the rule is triggered again at 5 elapsed seconds.
+* Then timer will be rescheduled for another 10 second countdown, so will execute the code within its block at 15 elapsed seconds.
+
+```javascript
+if (typeof this.timers['rescheduleTimerBlock'] === 'undefined' || this.timers['rescheduleTimerBlock'].hasTerminated()) {
+  this.timers['rescheduleTimerBlock'] = scriptExecution.createTimer(zdt.now().plusSeconds(10), function () {
+    logger.info('I am doing my job');
+    })
+} else {
+  this.timers['rescheduleTimerBlock'].reschedule(zdt.now().plusSeconds(10));
+}
+```
+
+### Cancel Timer
+
+![cancel-timer.png](images/blockly/blockly-cancel-timer.png)
+*Function*: Cancels the existing named timer, preventing code within the timer block from executing.
+
+### Timer is Active
+![timer-is-active](images/blockly/blockly-timer-is-active.png)
+
+*Function:* returns `true` if the timer itself is currently counting down, and the code within the timer will be eventually executed as scheduled. This block will return `false` if the timer doesn't exist, has been cancelled, or has already finished.
+
+### Timer is Running
+
+![timer-is-running](images/blockly/blockly-timer-is-running.png)
+*Function*: returns `true` if the code *within the timer block* is currently running - i.e. the countdown has finished, and the code within the timer block is currently executing. This event will usually only last a few milliseconds, and you’d be (un)lucky to catch it when your rule retriggers.
+
+### Timer has terminated
+
+![timer-terminated](images/blockly/blockly-timer-terminated.png)
+*Function*: returns `true` if the timer has finished its countdown, and the code within the timer block has finished running.
+
+### Comprehensive Timer Example
+
+The intent here is to trigger an alarm to the police but allow the timer to be reset in case a reset button has been pressed during the trigger time. The rule is triggered when the burglar-alarm-item or the reset-button-item is changed.
```

Comment:
Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.